File: src/main/java/exnihilocreatio/registries/types/Siftable.java
package exnihilocreatio.registries.types;

import exnihilocreatio.util.IStackInfo;
import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
public class Siftable {

    @Getter
    private IStackInfo drop;
    @Getter
    private float chance;
    @Getter
    private int meshLevel;
}

File: src/main/java/exnihilocreatio/registries/types/Siftable.kt
package exnihilocreatio.registries.types

import exnihilocreatio.util.ItemInfo

// TODO: MOVE TO API PACKAGE ON BREAKING VERSION CHANGE
data class Siftable(
        val drop: ItemInfo? = null,
        val chance: Float = 0.0f,
        val meshLevel: Int = 0
)
